database/connect.ts

```diff
@@ -31,8 +32,12 @@ function connectOneTimeToDatabase() {
       },
     });
   }
-  return globalThis.postgresSqlClient;
+
+  return (...sqlQuery: [TemplateStringsArray, ...any[]]) => {
+    headers();
+    return globalThis.postgresSqlClient!(...sqlQuery);
+  };
 }
 
 // Connect to PostgreSQL
-export const sql = connectOneTimeToDatabase();
+export const sql = connectOneTimeToDatabase() as ReturnType<typeof postgres>;
```

Comment:
When `globalThis.postgresSqlClient` is used with a regular function call instead of a tagged template literal